compiler/src/dotty/tools/dotc/core/TypeOps.scala

```diff
@@ -438,6 +438,8 @@ object TypeOps:
             tp.origin, fromBelow = variance > 0 || variance == 0 && tp.hasLowerBound)(using mapCtx)
           val lo1 = apply(lo)
           if (lo1 ne lo) lo1 else tp
+        case tp: LazyRef =>
```

Comment:
I think this should be `emptyRange` instead of `TypeBounds.empty`.

Reply:
Thanks for the tip, @odersky. I tried `emptyRange`, the test `tests/run/i2895a.scala` still fails. The problem is that `LazyRef` may leak into non-bounds places after `asSeenFrom`. It seems we need to stop earlier in the handling of bounds.